libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager.h

```diff
@@ -79,7 +79,7 @@ FOUNDATION_EXTERN NSString * const kDefaultCommunityName NS_SWIFT_NAME(SFDirecto
  @param components The additional path components to be added at the end of the directory (eg ['mybundle', 'common'])
  @return The path to the directory
  */
-- (nullable NSString*)directoryForUser:(SFUserAccount *)user scope:(SFUserAccountScope)scope type:(NSSearchPathDirectory)type components:(nullable NSArray *)components;
+- (nullable NSString*)directoryForUser:(nullable SFUserAccount *)user scope:(SFUserAccountScope)scope type:(NSSearchPathDirectory)type components:(nullable NSArray *)components;
```

Comment:
`user` should be `nonnull`, so should the directory returned. Are we making `user nullable` because the implementation falls back on `currentUser` if `nil` is passed in?

Reply:
Looking at the current implementation of this method, we can probably keep `user` parameter to be `nonnull`. However, if we want to annotate the directory returned to be `nonnull`, we would have to update the implementation, currently it does return `nil` in several logic branches:

```objc
if (nil == user.credentials.organizationId && scope != SFUserAccountScopeGlobal) {
        // do nothing
        return nil;
}

switch (scope) {
        case SFUserAccountScopeGlobal:
            return [self directoryForOrg:nil user:nil community:nil type:type components:components];
            
        case SFUserAccountScopeOrg:
            if (!user.credentials.organizationId) {
                [SFSDKCoreLogger w:[self class] format:@"Credentials missing for user"];
                return nil;
            }
            return [self directoryForOrg:user.credentials.organizationId user:nil community:nil type:type components:components];
            
        case SFUserAccountScopeUser:
            if (!user.credentials.organizationId || !user.credentials.userId) {
                [SFSDKCoreLogger w:[self class] format:@"Credentials missing for user"];
                return nil;
            }
            return [self directoryForOrg:user.credentials.organizationId user:user.credentials.userId community:nil type:type components:components];
            
        case SFUserAccountScopeCommunity:
            if (!user.credentials.organizationId || !user.credentials.userId) {
                [SFSDKCoreLogger w:[self class] format:@"Credentials missing for user"];
                return nil;
            }
            // Note: if the user communityId is nil, we use the default (internal) name for it.
            return [self directoryForOrg:user.credentials.organizationId user:user.credentials.userId community:user.credentials.communityId?:kDefaultCommunityName type:type components:components];
    }
```

Reply:
- `user` should be `nonnull`.
- Ideally, we should remove all the `userId` and `organizationId` checks and that would make `directory` `nonnull` as well, at that point. There's no reason a `user` object would be empty without a proper `organizationId`  or `userId`.

Could you make those changes, @AsceticMonk? cc: @trooper2013 @bbirman

Reply:
What about this check `scope != SFUserAccountScopeGlobal` at the beginning?

Reply:
Also, all cases converge to the method `[self directoryForOrg:user:community:type:components:]`, and this method has a `nullable` return type, so it makes sense that the return value of this method is `nullable` too, no?

Reply:
Alright, let's leave the return value as `nullable`, but `user` should be `nonnull` and those other checks can go.